src/StableBuffer.mo

```diff
@@ -74,6 +75,25 @@ module {
     };
   };
 
+  /// Removes the item at `index` from the buffer and returns it or `null` if 
+  /// `index >= size`.
+  /// All elements with index > `index` are shifted one position to the left.
+  /// Note: `index` is zero-based.
+  public func remove<X>(buffer: StableBuffer<X>, index : Nat) : ?X {
```

Comment:
I'd argue this should always return a value, and should trap otherwise.

Reply:
Sure, done now.